vertical-pod-autoscaler/hack/run-e2e-tests.sh

```
@@ -57,6 +57,21 @@ fi
 
 NUMPROC=${NUMPROC:-10}
 
+# Dump logs of all VPA component pods into ARTIFACTS so CI uploads them.
+function dump_vpa_logs {
+  mkdir -p "${ARTIFACTS}"
+  local component pod name
+  for component in admission-controller recommender updater; do
+    for pod in $(kubectl get pods -n "${ns}" -l "app.kubernetes.io/component=${component}" -o name 2>/dev/null); do
```

Shellcheck (0.11.0)

[warning] ns is referenced but not assigned.

(SC2154)